src/pudl/transform/ferc1.py

```diff
@@ -6092,6 +6093,63 @@ class OtherRegulatoryLiabilitiesTableTransformer(Ferc1AbstractTableTransformer):
 }
 
 
+def filter_for_freshest_data_xbrl(df: pd.DataFrame, primary_keys) -> pd.DataFrame:
+    """Get most updated values for each XBRL context.
+
+    An XBRL context includes an entity ID, the time period the data applies to, and
+    other dimensions such as utility type. Each context has its own ID, but they are
+    frequently redefined with the same contents but different IDs - so we identify
+    them by their actual content.
+
+    Each row in our SQLite database includes all the facts for one context/filing
+    pair.
+
+    If one context is represented in multiple filings, we take the most
+    recently-reported non-null value.
+
+    This means that if a utility reports a non-null value, then later
+    either reports a null value for it or simply omits it from the report,
+    we keep the old non-null value, which may be erroneous. This appears to
+    be fairly rare, affecting < 0.005% of reported values.
+    """
+
+    def __apply_diffs(
+        duped_groups: pd.core.groupby.DataFrameGroupBy,
+    ) -> pd.DataFrame:
+        """Take the latest reported non-null value for each group."""
+        return duped_groups.last()
```

Comment:
non-blocking: we had talked about in-lining this function, do you still want to do that?

Reply:
yea i think it's necessary to actuall run it over in the tests.